File: vdp-python/visa/test/vctc/test_consumer_rules.py
from visa.helpers.visa_api_client import VisaAPIClient
import json
import sys
import os
import unittest
if sys.version_info < (3, 0):
    import ConfigParser as parser
else:
    import configparser as parser
'''
@author: visa
'''

class TestConsumerRules(unittest.TestCase):

    config = parser.ConfigParser()
    config_path = os.path.abspath(os.path.join(os.path.dirname(os.path.dirname(__file__)),'..','configuration.ini'))
    config.read(config_path)
    
    def setUp(self):
        self.visa_api_client = VisaAPIClient()
        self.card_register_data = json.loads('''{
          "primaryAccountNumber": ''' + self.config.get('VDP','vctcTestPan') +'''
        }''')
    
    def test_register_a_card(self):
        base_uri = 'vctc/'
        resource_path = 'customerrules/v1/consumertransactioncontrols'
        response = self.visa_api_client.do_mutual_auth_request(base_uri + resource_path, self.card_register_data, 'Register a card call', 'post')
        self.assertTrue((str(response.status_code) == "200" or str(response.status_code) == "201"),"Register a card test failed")
        pass

File: visa-dev-samplecode-master/vdp-java/src/test/java/com/visa/vdp/vctc/TestConsumerRules.java
package com.visa.vdp.vctc;

import org.apache.http.client.methods.CloseableHttpResponse;
import org.testng.Assert;

import java.util.HashMap;

import org.apache.http.HttpStatus;
import org.testng.annotations.BeforeTest;
import org.testng.annotations.Test;

import com.visa.vdp.utils.VisaAPIClient;
import com.visa.vdp.utils.MethodTypes;
import com.visa.vdp.utils.Property;
import com.visa.vdp.utils.VisaProperties;

public class TestConsumerRules {

    String cardRegisterData;
    VisaAPIClient visaAPIClient;

    @BeforeTest(groups = "vctc")
    public void setup() {
        this.visaAPIClient = new VisaAPIClient();
        this.cardRegisterData = 
                "{"
                        + "\"primaryAccountNumber\": \""+ VisaProperties.getProperty(Property.VCTC_TEST_PAN) +"\""
              + "}";
    }

    @Test(groups = "vctc")
    public void testRegisterACard() throws Exception {
        String baseUri = "vctc/";
        String resourcePath = "customerrules/v1/consumertransactioncontrols";

        CloseableHttpResponse response = this.visaAPIClient.doMutualAuthRequest(baseUri + resourcePath, "Register A Card", this.cardRegisterData, MethodTypes.POST, new HashMap<String, String>());
        Assert.assertTrue((response.getStatusLine().getStatusCode() == HttpStatus.SC_OK || response.getStatusLine().getStatusCode() == HttpStatus.SC_CREATED));
        response.close();
    }

}
